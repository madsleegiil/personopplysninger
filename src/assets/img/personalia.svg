<?xml version="1.0" encoding="UTF-8"?>
<svg width="52px" height="62px" viewBox="0 0 52 62" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
    <!-- Generator: Sketch 53.2 (72643) - https://sketchapp.com -->
    <title>book-contact-2 Copy</title>
    <desc>Created with Sketch.</desc>
    <g id="Page-1" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
        <g id="book-contact-2-Copy">
            <path d="M6.74545455,0 L49.1454545,0 L49.1454545,12.7179487 L6.74545455,12.7179487 C3.19744342,12.7179487 0.321212121,9.87093892 0.321212121,6.35897436 C0.321212121,2.84700979 3.19744342,6.45137425e-16 6.74545455,0 Z" id="Rectangle" fill="#E7E9E9"></path>
            <path d="M50.4303032,10.1743589 L6.74545447,10.1743589 C4.62031506,10.1743589 2.89090899,8.46252298 2.89090899,6.35897426 C2.89090899,4.25542553 4.62031506,2.54358963 6.74545447,2.54358963 L50.4303032,2.54358963 C51.1395396,2.54358963 51.7151517,1.97382553 51.7151517,1.27179475 C51.7151517,0.572307567 51.1395396,-1.27999027e-07 50.4303032,-1.27999027e-07 L6.74545447,-1.27999027e-07 C3.20184233,-1.27999027e-07 0.321212009,2.85390758 0.321212009,6.35897426 C0.321212009,6.65911785 0.341769585,6.95417426 0.380315039,7.2415999 C0.341769585,7.3662358 0.321212009,7.49595888 0.321212009,7.63076913 L0.321212009,54.6871796 C0.321212009,58.1922463 3.20184233,61.046154 6.74545447,61.046154 L50.4303032,61.046154 C51.1395396,61.046154 51.7151517,60.4763899 51.7151517,59.7743591 L51.7151517,11.4461538 C51.7151517,10.7466666 51.1395396,10.1743589 50.4303032,10.1743589 Z" id="Fill/book-contact-2" fill="#D6897D" fill-rule="nonzero"></path>
            <path d="M37.5818183,48.3282051 L14.4545455,48.3282051 C13.7453091,48.3282051 13.169697,47.758441 13.169697,47.0564103 C13.169697,39.3010051 17.2966303,35.5542974 21.2976485,34.0001641 C19.4834424,32.6037333 18.3090909,30.4264205 18.3090909,27.9794872 C18.3090909,23.7723897 21.767903,20.3487179 26.0181818,20.3487179 C30.2684606,20.3487179 33.7272729,23.7723897 33.7272729,27.9794872 C33.7272729,30.4264205 32.5529214,32.6037333 30.7387152,34.0001641 C34.7397335,35.5517538 38.8666668,39.3010051 38.8666668,47.0564103 C38.8666668,47.758441 38.2910547,48.3282051 37.5818183,48.3282051 Z" id="Path" fill="#E7E9E9" fill-rule="nonzero"></path>
            <path d="M6.74545447,5.08717938 C6.0362181,5.08717938 5.46060598,5.65948707 5.46060598,6.35897426 C5.46060598,7.06100503 6.0362181,7.63076914 6.74545447,7.63076914 L47.8606062,7.63076914 C48.5698426,7.63076914 49.1454547,7.06100503 49.1454547,6.35897426 C49.1454547,5.65948707 48.5698426,5.08717938 47.8606062,5.08717938 L6.74545447,5.08717938 Z" id="Path" fill="#C6C2BF"></path>
        </g>
    </g>
</svg>