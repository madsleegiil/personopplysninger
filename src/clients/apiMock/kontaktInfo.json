{
  "mobiltelefonnummer": null,
  "epostadresse": null,
  "kanVarsles": null
}
