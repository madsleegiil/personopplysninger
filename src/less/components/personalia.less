.personalia-list {
  list-style: none;
  columns: 2;

  li {
    padding: 1rem;

    .title {
      font-weight: bold;
      display: block;
      white-space: nowrap;
      text-overflow: ellipsis;
      overflow: hidden;
    }
  }
}
