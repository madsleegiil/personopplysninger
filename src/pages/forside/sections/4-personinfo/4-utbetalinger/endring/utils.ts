import { OptionType } from "types/option";
import { Fields } from "calidation";
import {
  LAND_MED_BANKKODE,
  BIC,
} from "./utenlandsk-bankkonto/UtenlandsBankkonto";

import { LandOppslag } from "./landOppslag";

export const getCountryAlpha2 = (countryCode: string) =>
  LandOppslag.has(countryCode) ? LandOppslag.get(countryCode)?.alpha2 : "";

export const getIbanPrefixAlternatives = (countryCode: string): string[] => {
  return LandOppslag.has(countryCode)
    ? LandOppslag.get(countryCode)?.ibanPrefixAlternatives || []
    : [];
};

export const harValgtBic = (bankidentifier?: string) =>
  !!(bankidentifier && bankidentifier === BIC);

export const harValgtUSA = (land?: OptionType) =>
  !!(land && land.value === "USA");

export const brukerBankkode = (land?: OptionType) =>
  !!(land && LAND_MED_BANKKODE.includes(land.value));

export const validerBic = (fields: Fields) => {
  if (harValgtUSA(fields.land)) {
    return false;
  }

  if (brukerBankkode(fields.land)) {
    if (harUtfylt(fields.bickode) || !harUtfylt(fields.bankkode)) {
      return true;
    }
  }

  if (!brukerBankkode(fields.land)) {
    return true;
  }

  return false;
};

export const validerBankkode = (fields: Fields) => {
  if (harValgtUSA(fields.land)) {
    return true;
  }

  if (brukerBankkode(fields.land)) {
    if (harUtfylt(fields.bankkode) || !harUtfylt(fields.bickode)) {
      return true;
    }
  }

  return false;
};

export const harUtfylt = (value?: string) => (value ? true : false);

export const erLandIEuropa = (land: OptionType): boolean => {
  if (!land) {
    return false;
  }
  return !!(
    LandOppslag.has(land.value) && LandOppslag.get(land.value)?.isEurope
  );
};
