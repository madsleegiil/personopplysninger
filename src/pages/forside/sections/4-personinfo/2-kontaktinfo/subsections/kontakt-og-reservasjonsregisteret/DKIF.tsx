import React from "react";
import { Normaltekst } from "nav-frontend-typografi";
import { FormattedMessage } from "react-intl";
import ListElement from "components/listelement/ListElement";
import { KontaktInfo } from "types/kontaktInfo";
import Kilde from "components/kilde/Kilde";
import eksternLenkeIkon from "assets/img/Link.svg";
import { useStore } from "../../../../../../../store/Context";

interface Props {
  info: KontaktInfo;
}

const KontaktInformasjon = (props: Props) => {
  const { mobiltelefonnummer, epostadresse, kanVarsles } = props.info;
  const [{ locale }] = useStore();
  return mobiltelefonnummer || epostadresse || kanVarsles ? (
    <>
      <div className="underseksjon__beskrivelse">
        <Normaltekst>
          {kanVarsles ? (
            <FormattedMessage
              id="personalia.dkif.kanVarsles"
              values={{
                b: (text: string) => <b>{text}</b>,
              }}
            />
          ) : (
            <FormattedMessage id="personalia.dkif.kanIkkeVarsles" />
          )}
        </Normaltekst>
      </div>
      <ul className="list-column-2">
        <ListElement titleId="personalia.tlfnr" content={mobiltelefonnummer} />
        <ListElement titleId="personalia.email" content={epostadresse} />
      </ul>
      <Kilde
        kilde="personalia.source.dkif"
        lenke={`https://brukerprofil.difi.no/minprofil${
          locale === "en" ? "?locale=en" : ""
        }`}
        lenkeTekst="personalia.link.dkif.endre"
        lenkeType={"EKSTERN"}
        ikon={eksternLenkeIkon}
      />
    </>
  ) : (
    <>
      <div className="underseksjon__beskrivelse">
        <FormattedMessage
          id={"personalia.dkif.ingenData"}
          values={{
            b: (text: string) => <b>{text}</b>,
          }}
        />
      </div>
      <Kilde
        kilde="personalia.source.dkif"
        lenke="https://brukerprofil.difi.no/minprofil"
        lenkeTekst="personalia.link.dkif.leggtil"
        lenkeType={"EKSTERN"}
        ikon={eksternLenkeIkon}
      />
    </>
  );
};

export default KontaktInformasjon;
