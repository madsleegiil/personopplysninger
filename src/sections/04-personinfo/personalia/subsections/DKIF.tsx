import React, { Component } from "react";
import { Undertittel, Normaltekst } from "nav-frontend-typografi";
import { FormattedMessage } from "react-intl";
import Kilde from "../../../../components/kilde/Kilde";
import Environment from "../../../../utils/Environments";
import Error, { HTTPError } from "../../../../components/error/Error";
import Spinner from "../../../../components/spinner/Spinner";
import { fetchKontaktInfo } from "../../../../clients/apiClient";
import { KontaktInfo } from "../../../../types/kontaktInfo";
import KontaktInformasjon from "./KontaktInformasjon";

const { tjenesteUrl } = Environment();

type State =
  | { status: "LOADING" }
  | { status: "RESULT"; kontaktInfo: KontaktInfo }
  | { status: "ERROR"; error: HTTPError };

class DKIF extends Component<{}, State> {
  state: State = {
    status: "LOADING"
  };

  componentDidMount = () =>
    fetchKontaktInfo()
      .then((kontaktInfo: KontaktInfo) =>
        this.setState({
          status: "RESULT",
          kontaktInfo
        })
      )
      .catch((error: HTTPError) =>
        this.setState({
          status: "ERROR",
          error
        })
      );

  render = () => {
    return (
      <>
        <hr className="box__linje-bred" />
        <div className="underseksjon__overskrift">
          <Undertittel>
            <FormattedMessage id="personalia.dkif.overskrift" />
          </Undertittel>
        </div>
        <div className="underseksjon__beskrivelse">
          <Normaltekst>
            <FormattedMessage id="personalia.dkif.beskrivelse" />
          </Normaltekst>
        </div>
        {(() => {
          switch (this.state.status) {
            case "LOADING":
              return <Spinner />;
            case "RESULT":
              return <KontaktInformasjon info={this.state.kontaktInfo} />;
            case "ERROR":
              return <Error error={this.state.error} />;
          }
        })()}
        <Kilde
          tekst="personalia.source.dkif"
          lenkeTekst="personalia.link.dkif"
          href="https://brukerprofil.difi.no/minprofil"
        />
      </>
    );
  };
}

export default DKIF;
