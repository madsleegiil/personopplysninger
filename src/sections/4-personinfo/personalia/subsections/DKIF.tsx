import React, { PureComponent } from "react";
import { Undertittel, Normaltekst } from "nav-frontend-typografi";
import { FormattedMessage } from "react-intl";
import Error, { HTTPError } from "../../../../components/error/Error";
import Spinner from "../../../../components/spinner/Spinner";
import { fetchKontaktInfo } from "../../../../clients/apiClient";
import { KontaktInfo } from "../../../../types/kontaktInfo";
import KontaktInformasjon from "./KontaktInformasjon";

type State =
  | { status: "LOADING" }
  | { status: "RESULT"; kontaktInfo: KontaktInfo }
  | { status: "ERROR"; error: HTTPError };

class DKIF extends PureComponent<{}, State> {
  static state: State = {
    status: "LOADING"
  };

  componentDidMount = () => {
    if (DKIF.state.status !== "RESULT") {
      fetchKontaktInfo()
        .then((kontaktInfo: KontaktInfo) => {
          DKIF.state = {
            status: "RESULT",
            kontaktInfo
          };
        })
        .catch((error: HTTPError) => {
          DKIF.state = {
            status: "ERROR",
            error
          };
        })
        .then(() => this.forceUpdate());
    }
  };

  render = () => (
    <>
      <hr className="box__linje-bred" />
      <div className="underseksjon__overskrift">
        <Undertittel>
          <FormattedMessage id="personalia.dkif.overskrift" />
        </Undertittel>
      </div>
      <div className="underseksjon__beskrivelse">
        <Normaltekst>
          <FormattedMessage id="personalia.dkif.beskrivelse" />
        </Normaltekst>
      </div>
      {(() => {
        switch (DKIF.state.status) {
          case "LOADING":
            return <Spinner />;
          case "RESULT":
            return <KontaktInformasjon info={DKIF.state.kontaktInfo} />;
          case "ERROR":
            return <Error error={DKIF.state.error} />;
        }
      })()}
    </>
  );
}

export default DKIF;
