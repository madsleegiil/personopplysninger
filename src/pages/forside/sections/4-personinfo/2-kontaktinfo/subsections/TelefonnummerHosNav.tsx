import React, { useState } from "react";
import { Normaltekst, Undertittel } from "nav-frontend-typografi";
import { FormattedHTMLMessage } from "react-intl";
import { Tlfnr } from "types/personalia";
import leggTilIkon from "assets/img/LeggTil.svg";
import Kilde from "components/kilde/Kilde";
import Melding from "components/melding/Melding";
import EndreNummer from "./telefonnummer/EndreNummer";
import OpprettNummer from "./telefonnummer/OpprettNummer";
import { fjernMellorom } from "utils/formattering";
import driftsmeldinger from "driftsmeldinger";
import { AlertStripeAdvarsel } from "nav-frontend-alertstriper";

interface Props {
  tlfnr?: Tlfnr;
}

const TelefonnummerHosNav = (props: Props) => {
  const [opprett, settOpprett] = useState();
  const { tlfnr } = props;

  const onChangeSuccess = () => {
    settOpprett(false);
  };

  const onDeleteSuccess = () => {
    settOpprett(false);
  };

  const onLeggTil = () => settOpprett(!opprett);

  return (
    <>
      {driftsmeldinger.pdl && (
        <div style={{ paddingBottom: "1rem" }}>
          <AlertStripeAdvarsel>{driftsmeldinger.pdl}</AlertStripeAdvarsel>
        </div>
      )}
      <div className="underseksjon__header">
        <Undertittel>
          <FormattedHTMLMessage id="personalia.tlfnr.oveskrift" />
        </Undertittel>
      </div>
      {tlfnr && (tlfnr.telefonHoved || tlfnr.telefonAlternativ) ? (
        <div>
          {tlfnr.telefonHoved && (
            <EndreNummer
              prioritet={1}
              titleId="personalia.tlfnr.hoved"
              landskode={tlfnr.landskodeHoved}
              tlfnummer={fjernMellorom(tlfnr.telefonHoved)}
              onDeleteSuccess={onDeleteSuccess}
              onChangeSuccess={onChangeSuccess}
            />
          )}
          {tlfnr.telefonAlternativ && (
            <EndreNummer
              prioritet={2}
              titleId="personalia.tlfnr.alternativ"
              landskode={tlfnr.landskodeAlternativ}
              tlfnummer={fjernMellorom(tlfnr.telefonAlternativ)}
              onDeleteSuccess={onDeleteSuccess}
              onChangeSuccess={onChangeSuccess}
            />
          )}
        </div>
      ) : (
        <Melding meldingId="personalia.tlfnr.ingenData" />
      )}

      {!opprett && !(tlfnr && tlfnr.telefonHoved && tlfnr.telefonAlternativ) && (
        <button onClick={onLeggTil} className="tlfnummer__leggtil lenke">
          <span className="kilde__icon">
            <img src={leggTilIkon} alt="Ekstern lenke" />
          </span>
          <Normaltekst>
            <FormattedHTMLMessage id={"side.leggtil"} />
          </Normaltekst>
        </button>
      )}

      {opprett && (
        <OpprettNummer
          onCancelClick={() => settOpprett(false)}
          onChangeSuccess={onChangeSuccess}
          tlfnr={tlfnr}
        />
      )}

      <Kilde kilde="personalia.source.nav" lenkeType={"INGEN"} />
    </>
  );
};

export default TelefonnummerHosNav;
