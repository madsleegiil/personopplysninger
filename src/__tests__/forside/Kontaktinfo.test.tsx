import "@testing-library/jest-dom";
import renderer from "react-test-renderer";
import { IntlProvider } from "react-intl";
import nbMessages from "text/nb";
import kontaktInfo from "../../clients/apiMock/app/fetch/kontakt-info.json";
import personInfo from "../../clients/apiMock/app/fetch/person-info.json";
import DKIF from "../../pages/forside/sections/4-personinfo/2-kontaktinfo/subsections/kontakt-og-reservasjonsregisteret/DKIF";
import TelefonnummerHosNav from "../../pages/forside/sections/4-personinfo/2-kontaktinfo/subsections/TelefonnummerHosNav";
import { Tlfnr } from "types/personalia";
import { StoreProvider } from "../../store/Context";
import { extraValidators } from "../../utils/validators";
import { ValidatorsProvider } from "calidation";

describe("DKIF", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <StoreProvider>
          <IntlProvider locale={"nb"} messages={nbMessages}>
            <DKIF info={kontaktInfo} />
          </IntlProvider>
        </StoreProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("TelefonnummerHosNav", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <ValidatorsProvider validators={extraValidators}>
          <StoreProvider>
            <IntlProvider locale={"nb"} messages={nbMessages}>
              <TelefonnummerHosNav
                tlfnr={personInfo.personalia.tlfnr as unknown as Tlfnr}
              />
            </IntlProvider>
          </StoreProvider>
        </ValidatorsProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
