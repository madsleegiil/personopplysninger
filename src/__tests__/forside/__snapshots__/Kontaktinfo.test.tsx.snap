// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DKIF renders correctly 1`] = `
<DocumentFragment>
  <div
    class="underseksjon__beskrivelse"
  >
    <p
      class="navds-body-short"
    >
      Du har 
      <b>
        ikke
      </b>
       reservert deg mot elektronisk kommunikasjon.
    </p>
  </div>
  <dl
    class="list-column-2"
  >
    <div
      class="list-item"
    >
      <div
        class="listelement__header"
      >
        <dt>
          <p
            class="navds-label"
          >
            Telefon
          </p>
        </dt>
      </div>
      <div
        class="content "
      >
        <dd>
          12131415
        </dd>
      </div>
      <p
        class="navds-body-short"
      />
    </div>
    <div
      class="list-item"
    >
      <div
        class="listelement__header"
      >
        <dt>
          <p
            class="navds-label"
          >
            Språk/målform
          </p>
        </dt>
      </div>
      <div
        class="content "
      >
        <dd>
          Norsk
        </dd>
      </div>
      <p
        class="navds-body-short"
      />
    </div>
    <div
      class="list-item"
    >
      <div
        class="listelement__header"
      >
        <dt>
          <p
            class="navds-label"
          >
            E-post
          </p>
        </dt>
      </div>
      <div
        class="content "
      >
        <dd>
          test@test.com
        </dd>
      </div>
      <p
        class="navds-body-short"
      />
    </div>
  </dl>
  <div
    class="kilde__container"
  >
    <p
      class="navds-body-long"
    >
      <a
        class="navds-link kilde__lenke lenke"
        href="https://brukerprofil.difi.no/minprofil"
      >
        <span
          class="kilde__icon"
        >
          <img
            alt=""
            src="Link.svg"
          />
        </span>
        Endre i Kontakt- og reservasjonsregisteret 
      </a>
    </p>
    <div
      class="kilde__tekst"
    >
      <p
        class="navds-detail"
      >
        Kilde: Kontakt - og reservasjonsregisteret
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`TelefonnummerHosNav renders correctly 1`] = `
<DocumentFragment>
  <div
    class="underseksjon__header"
  >
    <h3
      class="navds-heading navds-heading--small"
    >
      Telefonnummer registrert hos NAV
    </h3>
  </div>
  <div>
    <div
      class="tlfnummer__rad"
    >
      <div
        class="tlfnummer__container"
      >
        <div
          class="tlfnummer__verdi"
        >
          <p
            class="navds-label"
          >
            Telefon 
          </p>
          <p
            class="navds-body-short"
          >
            <span>
              +47 
            </span>
            408 53 158
          </p>
        </div>
        <div
          class="tlfnummer__knapper"
        >
          <button
            aria-label="Endre telefonnummer"
            class="knapp-med-ikon navds-button navds-button--tertiary navds-button--medium"
            type="button"
          >
            <span
              aria-live="polite"
              class="navds-label"
            >
              <div
                class="tlfnummer__knapp-ikon"
              >
                <img
                  alt=""
                  src="Pencil.svg"
                />
              </div>
              <div
                class="tlfnummer__knapp-tekst"
              >
                Endre
              </div>
            </span>
          </button>
          <button
            aria-label="Slett telefonnummer"
            class="knapp-med-ikon navds-button navds-button--tertiary navds-button--medium"
            type="button"
          >
            <span
              aria-live="polite"
              class="navds-label"
            >
              <div
                class="tlfnummer__knapp-ikon"
              >
                <img
                  alt=""
                  src="Slett.svg"
                />
              </div>
              <div
                class="tlfnummer__knapp-tekst"
              >
                Slett
              </div>
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
  <button
    aria-label="Legg til telefonnummer"
    class="tlfnummer__leggtil knapp-med-ikon lenke navds-button navds-button--tertiary navds-button--medium"
  >
    <span
      aria-live="polite"
      class="navds-label"
    >
      <span
        class="kilde__icon"
      >
        <img
          alt=""
          src="LeggTil.svg"
        />
      </span>
      <p
        class="navds-body-short"
      >
        Legg til
      </p>
    </span>
  </button>
  <div
    class="kilde__container"
  >
    <div
      class="kilde__tekst"
    >
      <p
        class="navds-detail"
      >
        Kilde: NAV
      </p>
    </div>
  </div>
</DocumentFragment>
`;
